File: greedy/smallest-string-with-a-given-numeric-value/smallest-string-with-a-given-numeric-value.java
class Solution {
  public String getSmallestString(int n, int k) {
    char[] chars = new char[n];

    for (int i = n; i >= 1; i--) {
      int x = Math.min(k - (n - 1), 26);
      k -= x;
      n--;
      chars[i - 1] = (char) (x - 1 + 'a');
    }

    return new String(chars);
  }
}

File: greedy/smallest-string-with-a-given-numeric-value/smallest-string-with-a-given-numeric-value.ts
function getSmallestString(n: number, k: number): string {
  const chars: string[] = [];

  for (let i = n; i >= 1; i--) {
    let x: number = Math.min(k - (n - 1), 26);
    k -= x;
    n--;
    chars[i - 1] = String.fromCharCode(x - 1 + 97);
  }

  return chars.join('');
}
